ring/partition_ring_watcher.go

```diff
@@ -29,6 +30,10 @@ type PartitionRingWatcher struct {
 	numPartitionsGaugeVec *prometheus.GaugeVec
 }
 
+type PartitionRingWatcherDelegate interface {
+	OnPartitionRingChanged(oldRing, newRing *PartitionRingDesc)
```

Comment:
Can you add some comments here that explain what implementations should or should not do? e.g. don't modify the ring descriptions, must be concurrency safe, etc.